### DISCUSSION TITLE
[Feature] Auto add a watermark to photos on Immich

### DISCUSSION THREAD

**Discussion by Toanescphotographer** [immich-app/immich#7195] [Q&A] [answered]
I want to add a watermark to my photos on my Immich server, how can I do it?

**Comment by bo0tzz** [comment, marked as answer]
Immich doesn't ever change the original files, so you would need to set up some external tool for this.

**Reply by Toanescphotographer** [reply]
Thanks you!
i love it "Immich doesn't ever change the original files". Can you sugest me some ideas for "external tool"

**Reply by bo0tzz** [reply]
I've never done any watermarking so I don't have any tools to recommend, but I'm sure there are plenty out there.
